### ISSUE TITLE
Feature Request: add timeout parameter to the .fit() method

### ISSUE THREAD

**Issue by fingoldo** [catboost/catboost#2717]
Problem:

Adding the **timeout** parameter to the .fit() method, that should force the library to return best known solution found so far as soon as provided number of seconds since the start of training are passed, will allow to satisfy training SLAs, when a user has only a limited time budget to finish certain model training. Also, this will make possible fair comparison of different hyperparameters.

Reaching the timeout should have the same effect as reaching max iterations, maybe with additional warning and/or attribute set so that the training job's finishing reason is clear to the end user.

catboost version:
1.2.5
Operating System:
All
CPU: +
GPU: +

**Comment by jameslamb** [comment]
Linking the other reports you opened in other gradient boosting libraries:

* https://github.com/dmlc/xgboost/issues/10684
* https://github.com/microsoft/LightGBM/issues/6596

Please, let's keep these related conversations together so that we can try to coordinate across this ecosystem of similar libraries.
